src/mqueryfront/src/components/ActionShowMatchContext.js

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useRef, useEffect } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faLightbulb } from "@fortawesome/free-solid-svg-icons";
+import Draggable from "react-draggable";
+
+const useClickOutside = (ref, callback) => {
+    const handleClick = (event) => {
+        if (ref.current && !ref.current.contains(event.target)) {
+            // lose focus (higher z-index) only if other modal was clicked
+            const modals = document.querySelectorAll(".modal");
+            const wasClicked = (modal) => modal.contains(event.target);
+            if (Array.from(modals).some(wasClicked)) {
+                callback();
+            }
+        }
+    };
+
+    useEffect(() => {
+        document.addEventListener("click", handleClick);
+
+        return () => {
+            document.removeEventListener("click", handleClick);
+        };
+    });
+};
+
+function base64ToHex(str64) {
+    return (
+        atob(str64)
+            .split("")
+            .map(function (aChar) {
+                return ("0" + aChar.charCodeAt(0).toString(16)).slice(-2);
+            })
+            .join(" ")
+            .toUpperCase() + " "
+    );
+}
+
+function base64ToUtf8(str64) {
+    return (
+        atob(str64)
+            .split("")
+            .map(function (aChar) {
+                if (32 <= aChar.charCodeAt(0) && aChar.charCodeAt(0) < 127) {
+                    return aChar;
+                }
+                return ".";
+            })
+            .join(" ") + " "
```

Comment:
maybe remove the whitespace between letters? So `(" ")`. I understand that now spacing depends on this, but this should be solved anyway.